lib/test_suite.py

```diff
@@ -110,6 +111,17 @@ def __init__(self, suite_path, args):
                 lambda x: os.path.join(suite_path, x),
                 dict.fromkeys(self.ini[i].split())
                 if i in self.ini else dict())
+        if config.has_option("default", "fragile"):
+            fragiles = config.get("default", "fragile")
```

Comment:
Nit: I still don't get why not to use `self.ini.get('fragile', None)`. Since this dictionary is filled, it is used everywhere instead of `config` directly. Mix of `self.ini` and `config` accesses is a bit harder to read, IMHO.

Reply:
Not possible, because self.ini['fragile'] consists of dictionary instead of string needed for json (because of

self.ini.update(dict(config.items("default")))

below), and convert it to string mess it, check the following change:
```
diff --git a/lib/test_suite.py b/lib/test_suite.py
index d1faa46..cac377a 100644
--- a/lib/test_suite.py
+++ b/lib/test_suite.py
@@ -112,7 +112,8 @@ class TestSuite:
                 dict.fromkeys(self.ini[i].split())
                 if i in self.ini else dict())
         if config.has_option("default", "fragile"):
-            fragiles = config.get("default", "fragile")
+            fragiles = self.ini.get('fragile', None)
+            print("==================================== {}\n" . format(type(fragiles)))
             try:
                 self.fragile = json.loads(fragiles)
                 if 'tests' not in self.fragile:
```
got:
```
==================================== <type 'dict'>

Traceback (most recent call last):
  File "./test-run.py", line 219, in <module>
    status = main_parallel()
  File "./test-run.py", line 134, in main_parallel
    res = main_loop_parallel()
  File "./test-run.py", line 86, in main_loop_parallel
    task_groups = lib.worker.get_task_groups()
  File "/home/avtikhon/Workspaces/tarantool/test-run/lib/worker.py", line 70, in get_task_groups
    suites = find_suites()
  File "/home/avtikhon/Workspaces/tarantool/test-run/lib/worker.py", line 29, in find_suites
    for suite_name in sorted(suite_names)]
  File "/home/avtikhon/Workspaces/tarantool/test-run/lib/test_suite.py", line 118, in __init__
    self.fragile = json.loads(fragiles)
  File "/usr/lib/python2.7/json/__init__.py", line 339, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python2.7/json/decoder.py", line 364, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
TypeError: expected string or buffer
```

Reply:
And this dict is strange. Okay.